### ISSUE TITLE
Use of ReflectionConfigurationResources results in warning starting with GraalVM for JDK 21 (23.1)

### ISSUE THREAD

**Issue by zakkak** [neo4j/neo4j-java-driver#1498]
## Expected behavior

Building native-images with neo4j-java-driver should succeed without warnings.

## Actual behavior

Native builds print the following warning:

```
Warning: The option '-H:ReflectionConfigurationResources=META-INF/native-image/org.neo4j.driver/neo4j-java-driver/reflection-config.json' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
```

## More info

The option is passed in 
https://github.com/neo4j/neo4j-java-driver/blob/c5dc3b4b2347c3441e6dffb888518ebcb57bc9fc/driver/src/main/resources/META-INF/native-image/org.neo4j.driver/neo4j-java-driver/native-image.properties#L1

The reason this is required is because the configuration file is named `reflection-config.json` instead of `reflect-config.json`. Renaming the configuration file would result in GraalVM picking it up automatically without the need of passing `-H:ReflectionConfigurationResources`. See https://github.com/netty/netty/pull/13596 for a PR fixing the same issue in Netty.

cc @michael-simons

**Comment by zakkak** [comment]
Thank you @injectives!

**Comment by michael-simons** [comment]
Thank YOU @zakkak ! Much appreciated.

**Comment by injectives** [comment]
> Thank you @injectives!

Thank you @zakkak. The update is available in the new release: [5.15.0](https://github.com/neo4j/neo4j-java-driver/releases/tag/5.15.0).
